UnityProject/Assets/Scripts/Objects/PushPull.cs

```diff
@@ -491,6 +492,17 @@ public RightClickableResult GenerateRightClickOptions()
 			}
 		};
 	}
+	/// <summary>
+	/// Return true if the object causes gravity, otherwise false.
+	/// <returns>bool that represents if the object must cause gravity or not.</returns>
+	/// </summary>
+	public bool CausesGravity()
+    {
+		if (isNotPushable == true || floorDecal!= null)
```

Comment:
Not quite right, this causes lockers to stop when pushed out into space.
```suggestion
		if (!isNotPushable || floorDecal!= null)
```

The previous logic is - if an object is not pushable, it causes gravity, thus returning true from this method. Currently it would return false from this method if an object is not pushable. The suggested change fixes that.

Reply:
Thanks for the review @chairbender! You are right. I just wonder why does that happen on lockers since CausesGravity() method is only used by MetaTileMap.IsNoGravityAt() at the moment. Did you try using that method on other objects as well?